### PR TITLE
Create a base auth plugin to depend on instead of jupyverse

### PR DESCRIPTION
**Pull request by hbcarlos**
cc @davidbrochart

### CODE REVIEW

File: plugins/auth_base/fps_auth_base/__init__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+try:
+    User = auth["User"]
+    current_user = auth["current_user"]
+    update_user = auth["update_user"]
+    websocket_auth = auth["websocket_auth"]
+except KeyError:
+    raise RuntimeError(
+        "An auth plugin must be installed for fps-auth-base to run.",
+        "For instance: pip install fps-auth",
```
Comment (davidbrochart) [inline plugins/auth_base/fps_auth_base/__init__.py:15]:
```suggestion
        "An auth plugin must be installed, for instance: pip install fps-auth"
```

File: fps_cli_args.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[fps-auth]
+mode = "noauth"
```
Comment (davidbrochart) [inline fps_cli_args.toml:2]: This file should not be committed.

File: plugins/auth_base/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+[project.license]
+text = "BSD 3-Clause License"
+
+[project.urls]
+Homepage = "https://jupyter.org"
+
+[project.optional-dependencies]
+auth = [ "fps-auth ==0.0.41",]
+auth-fief = [ "fps-auth-fief ==0.0.41",]
+
```
Comment (davidbrochart) [inline plugins/auth_base/pyproject.toml:30]:
```suggestion
```

**Review by davidbrochart — CHANGES_REQUESTED** [review]
Thanks a lot Carlos, that looks good.

**Comment by hbcarlos** [comment]
Thanks, David! I added a commit with the changes.

File: pyproject.toml [hunk 4]
```diff
@@ -44,7 +45,7 @@ skip = [ "check-links", "check-manifest",]
 max-line-length = 100
 
 [tool.jupyter-releaser.options]
-python_packages = [ "plugins/auth:fps-auth", "plugins/auth_fief:fps-auth-fief", "plugins/contents:fps-contents", "plugins/frontend:fps-frontend", "plugins/jupyterlab:fps-jupyterlab", "plugins/kernels:fps-kernels", "plugins/lab:fps-lab", "plugins/nbconvert:fps-nbconvert", "plugins/retrolab:fps-retrolab", "plugins/terminals:fps-terminals", "plugins/yjs:fps-yjs", "plugins/login:fps-login", ".:jupyverse:fps-auth,fps-auth-fief,fps-contents,fps-jupyterlab,fps-kernels,fps-lab,fps-frontend,fps-nbconvert,fps-retrolab,fps-terminals,fps-yjs",]
+python_packages = ["plugins/auth:fps-auth-base", "plugins/auth:fps-auth", "plugins/auth_fief:fps-auth-fief", "plugins/contents:fps-contents", "plugins/frontend:fps-frontend", "plugins/jupyterlab:fps-jupyterlab", "plugins/kernels:fps-kernels", "plugins/lab:fps-lab", "plugins/nbconvert:fps-nbconvert", "plugins/retrolab:fps-retrolab", "plugins/terminals:fps-terminals", "plugins/yjs:fps-yjs", "plugins/login:fps-login", ".:jupyverse:fps-auth-base,fps-auth,fps-auth-fief,fps-contents,fps-jupyterlab,fps-kernels,fps-lab,fps-frontend,fps-nbconvert,fps-retrolab,fps-terminals,fps-yjs",]
```
Comment (davidbrochart) [inline pyproject.toml:48]: I forgot that one.
```suggestion
python_packages = ["plugins/auth_base:fps-auth-base", "plugins/auth:fps-auth", "plugins/auth_fief:fps-auth-fief", "plugins/contents:fps-contents", "plugins/frontend:fps-frontend", "plugins/jupyterlab:fps-jupyterlab", "plugins/kernels:fps-kernels", "plugins/lab:fps-lab", "plugins/nbconvert:fps-nbconvert", "plugins/retrolab:fps-retrolab", "plugins/terminals:fps-terminals", "plugins/yjs:fps-yjs", "plugins/login:fps-login", ".:jupyverse:fps-auth-base,fps-auth,fps-auth-fief,fps-contents,fps-jupyterlab,fps-kernels,fps-lab,fps-frontend,fps-nbconvert,fps-retrolab,fps-terminals,fps-yjs",]
```

**Review by davidbrochart — APPROVED** [review]

**Pull request merged**